particles/Native/Wasm/source/TestParticle.kt

```diff
@@ -107,17 +107,13 @@ class TestParticle : Particle() {
                  </table>""".trimIndent()
     }
 
-    private val data = Singleton { TestParticle_Data() }
-    private val res = Singleton { TestParticle_Res() }
-    private val info = Collection { TestParticle_Info() }
+    private val data = Singleton(this, "data", { TestParticle_Data() })
+    private val res = Singleton(this, "res", { TestParticle_Res() })
+    private val info = Collection(this, "info", { TestParticle_Info() })
```

Comment:
When a function/constructor has a lambda as its last argument, it's more idiomatic to place that lambda outside of the parens.

e.g.

```kotlin
private val data = Singleton(this, "data") { TestParticle_Data() }
```

Additionally, consider whether it would be worthwhile to make this into a property delegate. It would mean you wouldn't need to specify `this` and could become:

```kotlin
private val data by Singleton("data") { TestParticle_Data() }
```

Reply:
a) Ok that one's just weird (but I'm fine with changing to it).

b) That would be great! It's not all that clear how to make it work... specifically, how to set up the required `getValue` method.

Reply:
It doesn't look like that's going to work. The Singleton class needs the Particle reference during construction, and `thisRef` (which would be the Particle) only seems to be available via the `getValue`/`setValue` calls, which only occur when the field is accessed.